org.eclipse.xtext.generator/src/org/eclipse/xtext/generator/serializer/SerializerFragment.java

```diff
@@ -0,0 +1,184 @@
+/**
+ * Copyright (c) 2012, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.generator.serializer;
+
+import java.util.List;
+import java.util.Set;
+
+import org.eclipse.xpand2.XpandExecutionContext;
+import org.eclipse.xtext.Grammar;
+import org.eclipse.xtext.generator.BindFactory;
+import org.eclipse.xtext.generator.Binding;
+import org.eclipse.xtext.generator.IStubGenerating;
+import org.eclipse.xtext.generator.LanguageConfig;
+import org.eclipse.xtext.generator.Xtend2ExecutionContext;
+import org.eclipse.xtext.generator.Xtend2GeneratorFragment;
+import org.eclipse.xtext.generator.adapter.Generator2AdapterSetup;
+import org.eclipse.xtext.generator.parser.antlr.ex.wsAware.SyntheticTerminalAwareFragmentHelper;
+import org.eclipse.xtext.generator.terminals.SyntheticTerminalDetector;
+import org.eclipse.xtext.parser.antlr.AbstractSplittingTokenSource;
+import org.eclipse.xtext.serializer.ISerializer;
+import org.eclipse.xtext.serializer.impl.Serializer;
+import org.eclipse.xtext.serializer.sequencer.ISemanticSequencer;
+import org.eclipse.xtext.serializer.sequencer.ISyntacticSequencer;
+import org.eclipse.xtext.xtext.generator.CodeConfig;
+import org.eclipse.xtext.xtext.generator.serializer.SerializerFragment2;
+
+import com.google.common.collect.Lists;
+import com.google.inject.Binder;
+import com.google.inject.Inject;
+import com.google.inject.name.Names;
+
+/**
+ * @author Moritz Eyshold - Initial contribution and API
+ */
+@Deprecated
+public class SerializerFragment extends Xtend2GeneratorFragment
+		implements IStubGenerating, IStubGenerating.XtendOption {
+	@Inject
+	private SerializerGenFileNames names;
+
+	private boolean generateDebugData = false;
+
+	private boolean srcGenOnly = false;
+
+	/**
+	 * @since 2.8
+	 */
+	private boolean detectSyntheticTerminals = true;
```

Comment:
Off topic, why is there an `@since 2.8` on a private field? For documentation?

Reply:
no idea

Reply:
assume so